test/NuGet.Tests.Apex/NuGet.Tests.Apex/Utility/CommonUtility.cs

```diff
@@ -435,19 +435,30 @@ public static void UIInvoke(Action action)
 
         internal static ProjectTestExtension CreateAndInitProject(ProjectTemplate projectTemplate, SimpleTestPathContext pathContext, SolutionService solutionService, ILogger logger)
         {
-            logger.LogInformation("Creating solution");
-            solutionService.CreateEmptySolution("TestSolution", pathContext.SolutionRoot);
+            try
+            {
+                logger.LogInformation("Creating solution");
+                solutionService.CreateEmptySolution("TestSolution", pathContext.SolutionRoot);
+
+                logger.LogInformation("Adding project");
+                var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, ProjectTargetFramework.V46, "TestProject");
 
-            logger.LogInformation("Adding project");
-            var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, ProjectTargetFramework.V46, "TestProject");
+                logger.LogInformation("Saving all open and dirty files");
+                solutionService.SaveAll();
 
-            logger.LogInformation("Saving solution");
-            solutionService.Save();
+                logger.LogInformation("Saving solution");
+                solutionService.Save();
 
-            logger.LogInformation("Building solution");
-            project.Build();
+                logger.LogInformation("Building solution");
+                project.Build();
 
-            return project;
+                return project;
+            }
+            catch (Exception ex)
+            {
+                logger.LogInformation("[NuGet Fault in Apex]: " + ex.ToString());
```

Comment:
I think you need to `unwrap` the exception. 
See example in https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/ExceptionUtilities.cs#L30.

Reply:
The reason I didn't invoke the unwrap utils or write additional code for inner exceptions is that it appears to be the default for `ToString()`. Am I missing something?

> The default implementation of [ToString](https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=netframework-4.8) obtains the name of the class that threw the current exception, the message, the result of calling [ToString](https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=netframework-4.8) on the inner exception, and the result of calling [Environment.StackTrace](https://learn.microsoft.com/en-us/dotnet/api/system.environment.stacktrace?view=netframework-4.8).

https://learn.microsoft.com/en-us/dotnet/api/system.exception.tostring?view=netframework-4.8#remarks

Reply:
Checking my own work:

 ```
        static void Main(string[] args)
        {
            try
            {
                DoWork();
            }
            catch (Exception ex) 
            { 
                Console.WriteLine(ex);
            }

            Console.ReadLine();
        }

        private static void DoWork()
        {
            throw new Exception("Outer exception", new Exception("Inner exception"));
        }
```

![image](https://user-images.githubusercontent.com/49205731/225422226-f53956dc-a4c5-4c5b-b484-f7f3ec88f16a.png)

Reply:
AggregateException breaks that since there can be more than 1 exception, https://learn.microsoft.com/en-us/dotnet/api/system.aggregateexception.-ctor?view=netframework-4.7.2#overloads.

Reply:
I see. No, I expect it to be `SolutionException`.